pkg/providers/amifamily/bootstrap/bottlerocket.go

```diff
@@ -51,7 +55,7 @@ func (b Bottlerocket) Script() (string, error) {
 	}
 
 	if b.KubeletConfig != nil {
-		if len(b.KubeletConfig.ClusterDNS) > 0 {
+		if len(b.KubeletConfig.ClusterDNS) > 0 && b.ClusterCIDR == nil {
```

Comment:
Just for my learning, why we need to update this check to consider cluster CIDR?

Reply:
So you're correct I did not make this the most clear. If I am understanding correctly, in the event you don't have a custom CIDR to get the ClusterDNSIP you use the the default DNSIP from the KubeletConfig. So now that it supports custom CIDRs we would need to check for that. If that does not make sense let me know